### ISSUE TITLE
Refactor common BCHandling functionality into "helper" objects

### ISSUE THREAD

**Issue by pbauman** [grinsfem/grins#34]
Right now, I'm relying on inheritance for some common functionality, but there's also code duplication in places. It's getting a bit ugly, frankly. I think, perhaps, a better solution is to use helper objects. So, create a "fluids" helper objects, and "energy" helper object, etc. to put common functionality in one place, then each associated physics object can use these helper objects to encapsulate the common functionality.

So, in other words, prefer composition over inheritance.

**Comment by pbauman** [comment]
Dropping this. Taking a totally different approach to boundary conditions.
